### PR TITLE
Bumping supported versions of HF

### PR DESCRIPTION
**Pull request by donebydan**
Now supporting HF versions >=4.20.1
Minor changes needed: 
1. `use_amp` -> `use_cuda_amp` and `use_cpu_amp`
2. Additional `compute_loss_context_manager` in training script

### CODE REVIEW

**Review by wulu473 — COMMENTED** [review]
Just a few comments on pinning the version

File: src/dp_transformers/arguments.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from transformers import TrainingArguments as HfTrainingArguments
 from transformers import IntervalStrategy, logging
 from dataclasses import dataclass, field
-from datasets import tqdm_utils
+from datasets.utils import disable_progress_bar
```
[the remark this answers is not in the document]
Comment (donebydan) [reply src/dp_transformers/arguments.py:7]: This changed in the PR:
https://github.com/huggingface/datasets/pull/3897/files

File: setup.py [hunk 2]
```diff
@@ -28,10 +28,10 @@
         ]
     },
     install_requires=[
-        "transformers>=4.10.0",
+        "transformers>=4.20.1",
         "datasets<2.0.0",
```
Comment (donebydan) [inline setup.py:32]: @wulu473 the release of `datasets` with the `tqdm` changes is `2.0.0`. I want to be cautious, but I guess we need to demand `>=2.0.0` now. Thoughts?

File: src/dp_transformers/arguments.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (donebydan) [reply src/dp_transformers/arguments.py:7]: Which was included in the following release:
https://github.com/huggingface/datasets/releases/tag/2.0.0

Please see comment in `setup.py`

**Review by wulu473 — APPROVED** [review]

**Pull request merged**